### PR TITLE
move the `/account` UI to LiveViews

### PR DESCRIPTION
**Pull request by joshk**

### CODE REVIEW

File: lib/nerves_hub_web/router.ex [hunk 1]
```diff
@@ -205,70 +205,57 @@ defmodule NervesHubWeb.Router do
...
+        on_mount: [
+          NervesHubWeb.Mounts.AccountAuth,
+          NervesHubWeb.Mounts.CurrentPath,
+          NervesHubWeb.Mounts.FetchOrg,
+          NervesHubWeb.Mounts.FetchOrgUser
+        ] do
+        live("/", Live.Org.Products, :index)
+        live("/new", Live.Org.Products, :new)
+        live("/settings/keys", Live.Org.SigningKeys, :index)
+        live("/settings/keys/new", Live.Org.SigningKeys, :new)
```
Comment (jjcarstens) [inline lib/nerves_hub_web/router.ex:233]: I actually liked all of this being outside of a scope with the full path in the `live` call. Makes it easier to find what I'm used to seeing in url

What's the motivation here to move it into a scope?

File: lib/nerves_hub/accounts/user_token.ex [hunk 2]
```diff
@@ -18,16 +18,17 @@ defmodule NervesHub.Accounts.UserToken do
...
   def create_changeset(%User{} = user, attrs) do
     %__MODULE__{user_id: user.id}
-    |> Changeset.cast(attrs, [:note, :user_id])
-    |> Changeset.put_change(:token, generate(user))
-    |> Changeset.validate_required([:token, :note, :user_id])
-    |> Changeset.validate_format(:token, ~r/^nhu_[a-zA-Z0-9]{36}$/)
-    |> Changeset.foreign_key_constraint(:user_id)
-    |> Changeset.unique_constraint(:token)
+    |> cast(attrs, [:note, :user_id])
+    |> update_change(:note, &trim/1)
```
[the remark this answers is not in the document]
Comment (joshk) [reply lib/nerves_hub/accounts/user_token.ex:22]: its also to trim extra spaces, eg. "Hello     Jon" -> "Hello Jon"

I know I'm being a bit over the top here, but, well, meh. :D

File: lib/nerves_hub/accounts/user_token.ex [hunk 3]
```diff
@@ -1,7 +1,7 @@
 defmodule NervesHub.Accounts.UserToken do
   use Ecto.Schema
+  import Ecto.Changeset
```
[the remark this answers is not in the document]
Comment (joshk) [reply lib/nerves_hub/accounts/user_token.ex:3]: I'm open to using `only`. We were being inconsistent in our use of Changeset, so I followed suit with how other modules were doing it. We can switch them all over but could you please create an issue for tracking this?

File: lib/nerves_hub_web/live/account.ex [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  defp apply_action(socket, :delete, _params) do
+    socket
+    |> page_title("Delete Account")
+    |> assign(:form, to_form(%{}))
+    |> render_with(&delete_account_template/1)
+  end
+
+  @impl true
+  def handle_event("update", %{"user" => params}, socket) do
+    cleaned = whitelist(params, [:current_password, :password, :username, :email, :orgs])
```
[the remark this answers is not in the document]
Comment (joshk) [reply lib/nerves_hub_web/live/account.ex:36]: honestly, I don't love this either, and I wanted to tidy this up, but I just moved over the code from the old Controller way to the new LiveView way with the goal of fixing this later.

File: lib/nerves_hub_web/router.ex (hunk 1 above)
Comment (joshk) [reply lib/nerves_hub_web/router.ex:233]: I can switch it back.

File: lib/nerves_hub_web/live/account.ex [hunk 5]
```diff
@@ -0,0 +1,69 @@
+defmodule NervesHubWeb.Live.Account do
+  use NervesHubWeb, :updated_live_view
+
+  alias NervesHub.Accounts
+
+  embed_templates("account_templates/*")
+
+  @impl true
```
[the remark this answers is not in the document]
Comment (joshk) [reply lib/nerves_hub_web/live/account.ex:8]: sure, I can update these.

File: lib/nerves_hub/accounts/user_token.ex [hunk 6]
```diff
@@ -18,16 +18,17 @@ defmodule NervesHub.Accounts.UserToken do
...
+    |> cast(attrs, [:note, :user_id])
+    |> update_change(:note, &trim/1)
+    |> put_change(:token, generate(user))
+    |> validate_required([:token, :note, :user_id])
+    |> validate_format(:token, ~r/^nhu_[a-zA-Z0-9]{36}$/)
+    |> foreign_key_constraint(:user_id)
+    |> unique_constraint(:token)
   end
 
   def update_chagneset(%__MODULE__{} = token, attrs) do
```
[the remark this answers is not in the document]
Comment (joshk) [reply lib/nerves_hub/accounts/user_token.ex:30]: hahahaha, amazing!

**Review by jjcarstens — APPROVED** [review]

**Pull request merged**